tests/toolchain/build-pass/using_std_array.cpp

```diff
@@ -0,0 +1,30 @@
+/*
+ * Regression test for https://github.com/EOSIO/eosio.cdt/issues/558
+ *
+ * Verifies that a class/function can be used from the std namespace
+ */
+
+#include <eosio/eosio.hpp>
+#include <eosio/print.hpp>
+#include <array>
+
+using std::array;
+using namespace eosio;
+
+class[[eosio::contract("hello")]] hello : public contract
```

Comment:
It's better to rename this contract to be the same as the file name (i.e. `using_std_array`), otherwise due to CDT change https://github.com/EOSIO/eosio.cdt/pull/1037, using `eosio-cpp` with this file as input and no options like `-o hello.wasm` will not generate the ABI file. 
Also, if this PR is more closely related with ABI generation, it can be put under `abigen-pass` so that the generated ABI can be checked.

Reply:
Sure, changed to using_std_array.